test/aws_s3_tests.jl

```diff
@@ -0,0 +1,225 @@
+@testitem "Basic AWS S3 usage" setup=[InitializeObjectStore] begin
+using CloudBase.CloudTest: Minio
+using RustyObjectStore: get!, put, AwsConfig, ClientOptions
+
+# For interactive testing, use Minio.run() instead of Minio.with()
+# conf, p = Minio.run(; debug=true, public=false); atexit(() -> kill(p))
+Minio.with(; debug=true, public=false) do conf
```

Comment:
Can we parameterize these tests so that they don't have to be duplicated? Maybe like how we have it in the pager where the actual test is in a function and then we call that inside Minio and Azurite blocks.

Reply:
I dislike the duplication here too, in the specific case of these tests, I think we should be good with a single implementation.

But there are other tests just as duplicated as these that actually have different errors for different store backends which makes it a little cumbersome to make them generic.

Do you think it is worth it to pursue removing all duplication or just part of it is enough?

Reply:
I think just the easy to dedup happy path tests are enough for now.